### PR TITLE
home page added

### PR DESCRIPTION
**Pull request by chanchal-panpaliya**
1)All needed components added - card ,screen loader , toast , modal , frame , header , footer , filter , slider , card slider ,toaster , etc.
2) Auth  with all api services 
3) Home page .
4) Login , Regis , Logout , Forgot password page added.

Screens are -
image
image
image
image
image
image

### CODE REVIEW

File: src/web-app/Component/AddNote/AddNote.jsx [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import { useState } from 'react';
+import './AddNote.css';
+const AddNote=({videoId,data})=>{
+    const [text,settext] = useState(data.note)
+
+    const handleNote=()=>{
+
+    }
```
Comment (Mohasin-Haque) [inline src/web-app/Component/AddNote/AddNote.jsx:8]: Why empty function?

File: src/web-app/Component/Card/Card.css [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+    margin-left: 5px;
+    margin-right: 5px;
+    box-shadow: 5px 5px 1.25rem 0px rgb(0 0 0 / 10%);
+    position: relative;
+    background: var(--BG_GRADIENT_L2);
+    transition: transform .3s; 
+}
+
+
+
```
Comment (Mohasin-Haque) [inline src/web-app/Component/Card/Card.css:58]: You can remove these spaces for sanity

File: src/web-app/Component/Card/Card.css [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+  
+  /* Small devices (portrait tablets and large phones, 600px and up) */
+  @media only screen and (min-width: 600px) {
+
+  }
+  
+  /* Medium devices (landscape tablets, 768px and up) */
+  @media only screen and (min-width: 768px) {
+
+  }
```
Comment (Mohasin-Haque) [inline src/web-app/Component/Card/Card.css:221]: I guess you have left this to add responsiveness

File: src/web-app/Component/Card/Card.jsx [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+                            <label className='--background video-label-hover'> Add palylist list </label>
+                            <label className='--background video-label-hover'> Watch Later </label>
+                        </div>  
+                    </div>
+                   
+                            {/* <div className="video-card-title"> {data.title} </div> */}
+                            {/* <div className="flex-row col-gap-2rem flex-justify-content-center video-card-description">
+                                 {data.description}
+                            </div> */}
+               </div>
```
Comment (Mohasin-Haque) [inline src/web-app/Component/Card/Card.jsx:100]: You can remove this before merge if its not needed

File: src/web-app/Component/Card/Card.jsx [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+                                {
+                                    token?
+                                      check?
+                                        <button className='--background video-label-hover' onClick={()=>handler_removeWatchLater(token, removedwatchlist, data._id)}>removed Watch</button>
+                                        :
+                                        <button className='--background video-label-hover' onClick={()=>handler_addWatchLater(token, addwatchlist, data)}>add Watch</button>
+                                       :
+                                       <button className='--background video-label-hover' onClick={()=>{setmodal(!ismodal)}}>without login</button>  
+                                }
+                            </div>
```
Comment (Mohasin-Haque) [inline src/web-app/Component/Card/Card.jsx:138]: same here also.

File: src/web-app/Component/Card/index.css [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+}
+
+.watchlist-btn::before{
+    content: '';
+    position: absolute;
+    top: 0;
+    left: -5px;
+    /* height: 35px;
+    width: 35px; */
+    background-size: cover;
```
Comment (Mohasin-Haque) [inline src/web-app/Component/Card/index.css:62]: same here

File: src/web-app/Pages/Auth/Register.jsx [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+    const [retypedPassword, setRetypedPassword] = useState("");
+    const [error, setError] = useState("");
+    const [lastname,setLastName]=useState("");
+    const [termsAndConditions,settermsAndConditions]=useState(false);
+    const [hideshowpassword,sethideshowpassword]=useState(false);
+    const [hideshow_rety_Pass,Sethideshow_rety_Pass]=useState(false);
+
+
+    const validateemail=(e)=>{
+        const emailRegex = /\S+@\S+\.\S+/;
```
Comment (Mohasin-Haque) [inline src/web-app/Pages/Auth/Register.jsx:21]: What this do??

**Review by Mohasin-Haque — COMMENTED** [review]
* Nice folder structure.
* Good coding practices.
* Left some suggestions. You can work on it.

**Comment by Mohasin-Haque** [comment]
Please try to make small PR

File: src/web-app/Component/AddNote/AddNote.jsx [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+
+    return(
+       <div>
```
Comment (sohan9819) [inline src/web-app/Component/AddNote/AddNote.jsx:11]: If `div` is not needed you can you react fragments here `<></>`

File: src/web-app/Component/Card/CardHome.jsx [hunk 9]
```diff
@@ -0,0 +1,61 @@
+import './index.css';
+import { useEffect, useState ,useContext} from 'react';
+import { Link } from 'react-router-dom';
+import VideoContext from 'web-app/Context/video/VideoContext';
+import { useAuth } from 'web-app/Context/login/AuthContext';
```
Comment (sohan9819) [inline src/web-app/Component/Card/CardHome.jsx:5]: I guess you can use barrel import here

File: src/web-app/Component/Card/CardHome.jsx [hunk 10]
```diff
@@ -0,0 +1,61 @@
...
+    </div>
+    {
+        ismodal ? <Auth_Modal modalClose={()=>setmodal(false)} /> :null
+    }
+        </>
+   
+    )
+}
+
+const CardPlaylsitContainer=({data})=>{
```
Comment (sohan9819) [inline src/web-app/Component/Card/CardHome.jsx:46]: Here also you can keep this components in separate file

File: src/web-app/Component/Card/index.css [hunk 11]
```diff
@@ -0,0 +1,68 @@
...
+.card-body{
+    opacity: 0;
+    width: 100%;
+    height: 50%;
+    position: absolute;
+    bottom: 0;
+    left: 0;
+    z-index: 1;
+    background: linear-gradient(to bottom, rgba(4, 8, 15, 0), #192133 100%);
+    padding: 10px;
```
Comment (sohan9819) [inline src/web-app/Component/Card/index.css:36]: try to avoid `px` units and use `em` or `rem`

File: src/web-app/Component/Footer/Footer.jsx [hunk 12]
```diff
@@ -0,0 +1,77 @@
...
+                       <div class="footer-col-3  --background"> 
+                          <h3>  </h3>
+                          <ul>
+                              <li>  </li>
+                              <li> </li>
+                              <li> </li>
+                              <li>  </li>
+                          </ul>
+                       </div>
+                   </div>
```
Comment (sohan9819) [inline src/web-app/Component/Footer/Footer.jsx:68]: Remove if you don't need this

File: src/web-app/Component/Frame/Frame.css [hunk 13]
```diff
@@ -0,0 +1,81 @@
+.carousel-container{
+    position: relative;
+    width: 100%;
+    height: 450px;
```
Comment (sohan9819) [inline src/web-app/Component/Frame/Frame.css:4]: Avoid `px`

File: src/web-app/Service/service.jsx [hunk 14]
```diff
@@ -0,0 +1,372 @@
...
+      }catch (error) {
+        console.log(error);
+    }
+}
+
+//movie data
+export async function fetch_movie_Data() {
+    try {
+       let data = [];
+      data = await axios.get("/api/videos").then(res=> res.data.videos)
```
Comment (sohan9819) [inline src/web-app/Service/service.jsx:364]: Can use object de structuring ``` const {data} =await axios.get("/api/videos")```

**Review by sohan9819 — APPROVED** [review]
Everything looks good . Host the project so that we cab see a preview and try to avoid doing big prs . Just a suggestion to break you code into different files instead of putting things together in a single file . Added few comments and suggestion . 👍

File: src/web-app/Component/AddNote/AddNote.jsx (hunk 1 above)
Comment (chanchal-panpaliya) [reply src/web-app/Component/AddNote/AddNote.jsx]: done

File: src/web-app/Component/Card/Card.css (hunk 2 above)
Comment (chanchal-panpaliya) [reply src/web-app/Component/Card/Card.css:58]: done

File: src/web-app/Component/Card/Card.css (hunk 3 above)
Comment (chanchal-panpaliya) [reply src/web-app/Component/Card/Card.css]: done

File: src/web-app/Component/Card/Card.jsx (hunk 4 above)
Comment (chanchal-panpaliya) [reply src/web-app/Component/Card/Card.jsx]: done

File: src/web-app/Component/Card/Card.jsx (hunk 5 above)
Comment (chanchal-panpaliya) [reply src/web-app/Component/Card/Card.jsx]: done

File: src/web-app/Component/Card/index.css (hunk 6 above)
Comment (chanchal-panpaliya) [reply src/web-app/Component/Card/index.css]: done

File: src/web-app/Pages/Auth/Register.jsx (hunk 7 above)
Comment (chanchal-panpaliya) [reply src/web-app/Pages/Auth/Register.jsx:21]: email id validation but i'm not using it .

File: src/web-app/Component/AddNote/AddNote.jsx (hunk 8 above)
Comment (chanchal-panpaliya) [reply src/web-app/Component/AddNote/AddNote.jsx:7]: here ,  i want div tag for new line.

File: src/web-app/Component/Card/Card.jsx [hunk 15]
```diff
@@ -0,0 +1,152 @@
...
+                                >
+                                </div>
+                        </Link>  
+                </div>
+          </>  
+    )
+}
+
+
+const VideoListCard=({data})=>{
```
[the remark this answers is not in the document]
Comment (chanchal-panpaliya) [reply src/web-app/Component/Card/Card.jsx:29]: I separate card component file as per the functionality-playlist, history, etc.

File: src/web-app/Component/Card/CardHome.jsx (hunk 9 above)
Comment (chanchal-panpaliya) [reply src/web-app/Component/Card/CardHome.jsx:5]: ok

File: src/web-app/Component/Card/index.css (hunk 11 above)
Comment (chanchal-panpaliya) [reply src/web-app/Component/Card/index.css:36]: As per the Vikram Sir lec, -> less than 16px, we can use px.

**Pull request merged**